src/MessagePack.GeneratorCore/CodeAnalysis/TypeCollector.cs

```diff
@@ -635,7 +635,8 @@ private ObjectSerializationInfo GetObjectInfo(INamedTypeSymbol type)
                     }
 
                     var customFormatterAttr = item.GetAttributes().FirstOrDefault(x => x.AttributeClass.ApproximatelyEqual(this.typeReferences.MessagePackFormatterAttribute))?.ConstructorArguments[0].Value as INamedTypeSymbol;
-                    var member = new MemberSerializationInfo(true, isWritable, isReadable, hiddenIntKey++, item.Name, item.Name, item.Type.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat), item.Type.ToDisplayString(BinaryWriteFormat), customFormatterAttr?.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat));
+                    var stringKey = item.GetAttributes().FirstOrDefault(x => x.AttributeClass.ApproximatelyEqual(this.typeReferences.KeyAttribute))?.ConstructorArguments[0].Value as string ?? item.Name;
```

Comment:
I expect this to throw when `[Key]` (without any argument) is encountered. Can you include a length check for `ConstructorArguments` before indexing into `[0]`?

Reply:
Can do. In the case where there are no args - should it default back to `item.Name`?